### ISSUE TITLE
Choosing "Clear" doesn't remove the annotation?

### ISSUE THREAD

**Issue by jamiechong** [cositehq/sanity-plugin-simpler-color-input#3]
Is it intended that clicking "Clear" doesn't actually remove the annotation? I end up in a state where my text is marked up, but without any color. 

![Screenshot 2023-12-28 at 10 19 59 AM](https://github.com/cositehq/sanity-plugin-simpler-color-input/assets/981664/8fff1b42-d8f4-41d4-a0eb-ca7c84c67583)
![Screenshot 2023-12-28 at 10 20 21 AM](https://github.com/cositehq/sanity-plugin-simpler-color-input/assets/981664/0887dff8-45cc-4448-bdb4-f42460321f44)
![Screenshot 2023-12-28 at 10 23 52 AM](https://github.com/cositehq/sanity-plugin-simpler-color-input/assets/981664/9c070055-2630-49d5-ae35-86cabbcc6d30)

Edit: for some reason seems like screenshots didn't upload correctly. Trying again.

**Comment by jamiechong** [comment]
seems related to how GH handles image uploads to tickets. Please comment back if you want me to share this another way. But hopefully you understand the flow I'm describing.

**Comment by AlyssaKirstine** [comment]
Hi @jamiechong! This is intentional. Clicking the Clear button clears the color the user selected, but to remove the annotation itself, you'll need to click outside of the Tex color popup and click the trash icon (see below screenshot).


![image](https://github.com/cositehq/sanity-plugin-simpler-color-input/assets/22719577/ebb2b7a1-ed98-44ec-87f4-3ba44802d539)

To prevent an empty color in my projects, I make the text color type required by setting it up like so under annotations:

```
{
    type: 'textColor',
    validation: (rule: Rule) => rule.required(),
}
```
Then the user will get an error message saying the field is required unless they delete it.
![image](https://github.com/cositehq/sanity-plugin-simpler-color-input/assets/22719577/a74bfb7b-823b-428b-817b-1f0b2b949317)

Does this make sense? Happy to also consider your thoughts if you think there's a better solution.

**Comment by jamiechong** [comment]
Thanks - I'll try using the validation Rule so we don't end up in a weird state where the annotation is added, but no color is set. I wonder what the use case is for even having a Clear button. Imo there is either a color or there is not. "clearing" it to be empty doesn't make sense to me, which is why I expected it to behave like the delete button.

**Comment by jamiechong** [comment]
I just tried the validation: required approach, but feel the UX is even more confusing because it still let's me clear the color, but then I don't discover the color is required until after I've cleared it. 

Possibly we can add an option to this plugin that let's us remove the Clear button?

**Comment by AlyssaKirstine** [comment]
Good point @jamiechong. Initially, I added the clear button when the color picker was built to be used outside the portable text editor (as an inline field for a document). In that case, you need a way to clear the color in cases where it is optional and you want to leave the field blank.

I agree that it is confusing when used in the portable text editor, however. I will look at the code when I get a chance to see what I can do to clear up the confusion!

**Comment by AlyssaKirstine** [comment]
Hey @jamiechong I just released [v2.1.0](https://github.com/cositehq/sanity-plugin-simpler-color-input/releases/tag/v2.1.0) of this plugin which automatically hides the clear button when the field is marked as required. This is how the color input for other CMSs like Contentful behave, so I thought it made sense to do this instead of adding an additional option to the plugin config.

Cheers,
Alyssa

**Comment by jamiechong** [comment]
amazing @AlyssaKirstine  - we'll give this a try. thank you!
